yang/ietf-l2vpn-ntw.yang

```diff
@@ -140,16 +140,91 @@ module ietf-l2vpn-ntw {
   /* Typedefs */
     /* 
 
+  typedef esi-type {
+    type enumeration {
+      enum TYPE_0_OPERATOR_CONFIGURED {
+        value 0;
+        description
+          "Type 0 indicates the value must be provided.";
+      }
+      enum TYPE_1_LACP_BASED {
+        value 1;
+        description
+          "Type 1 when IEEE 802.1AX LACP is used between the PEs and CEs, 
+          this ESI type indicates an auto-generated ESI value
+          determined from LACP.";
+      }
+      enum TYPE_2_BRIDGE_PROTOCOL_BASED {
+        value 2;
+        description
+          "Type 2 the ESI Value is auto-generated and determined based 
+          on the Layer 2 bridge protocol";
+      }
+      enum TYPE_3_MAC_BASED {
+        value 3;
+        description
+          "Type 3 this type indicates a MAC-based ESI Value that can be auto-generated 
+          or configured by the operator.";
+      }
+      enum TYPE_4_ROUTER_ID_BASED {
+        value 4;
+        description
+          "Type 4 this type indicates a router-ID ESI Value that can be auto-generated
+          or configured by the operator.";
+      }
+      enum TYPE_5_AS_BASED {
+        value 5;
+        description
+          "Type 5 5his type indicates an Autonomous System (AS)-based ESI Value that can 
+          be auto-generated or configured by the operator.";
+      }
+    }
+    description
+      "T-(ESI Type) is a 1-octet field (most significant octet) that
+      specifies the format of the remaining 9 octets (ESI Value).";
+    reference
+      "RFC 7432: BGP MPLS-Based Ethernet VPN page-16";
+  }
+
   typedef ccm-priority-type {
     type uint8 {
       range "0..7";
     }
     description
       "A 3-bit priority value to be used in the VLAN tag,
-       if present in the transmitted frame.";
+      if present in the transmitted frame.";
   }
+
   /* Identities */
 
+  identity df-election-methods {
+    description 
+    "Base Identity DF election method";
+  }
+
+  identity default {
+    base df-election-methods;
+    description 
+      "The default DF election method";
+    reference 
+      "draft-rabadan-bess-evpn-pref-df";
+  }
+
+  identity highest-random-weight {
+    base df-election-methods;
+    description 
+      "The highest random weight (HRW) method";
+    reference 
+      "draft-mohanty-bess-evpn-df-election";
+  }
+  identity preference {
+    base df-election-methods;
+    description 
+      "The preference based method";
+    reference 
```

Comment:
This is still a draft: draft-ietf-bess-evpn-pref-df. Only RFC need to be cited. Please remove the reference statement.